firmware/test/consoleFnTests.cpp

```diff
@@ -0,0 +1,338 @@
+#define CATCH_CONFIG_MAIN
+#include "base.h"
+#include "../src/consoleFunctions.cpp"
+#include "../src/consoleFunctions.h"
+#include "../src/flashAddresses.h"
+
+SCENARIO( "Turn_Debugging_Publishes_On_Off", "[toggle debug flag]" ) {
+
+    GIVEN( "A false debug flag" ) {
+        stateMachineDebugFlag = false;
+
+        WHEN( "the function is called with 1" ) {
+            toggle_debugging_publishes("1");
+
+            THEN( "debug flag should be set to true" ) {
+                REQUIRE( stateMachineDebugFlag == true );
+            }
+        }
+
+        WHEN( "the function is called with 0" ) {
+            toggle_debugging_publishes("0");
+
+            THEN( "debug flag should remain false" ) {
+                REQUIRE( stateMachineDebugFlag == false );
+            }
+        }
+
+        WHEN( "the function is called with neither 0 nor 1" ) {
+            int returnVal = toggle_debugging_publishes("invalid Value");
+
+            THEN( "debug flag should remain false" ) {
+                REQUIRE( stateMachineDebugFlag == false );
+            }
+
+            THEN( "the return value should be -1" ) {
+                REQUIRE( returnVal == -1 );
+            }
+
+        }
+    }
+        GIVEN( "A true debug flag" ) {
```

Comment:
All these tests look great. But could you please run through this file the check that the indentation is consistent and that there aren't any extra newlines? Thanks!